### ISSUE TITLE
rend2: skybox edges visible

### ISSUE THREAD

**Issue by bugfood** [iortcw/iortcw#86]
Hi,

When I enable `cl_renderer rend2`, the edges (and vertices) of the sky box are visible. It's hard to get a good screenshot to illustrate this, but it's very noticeable during player movement. Here's the best screenshot I have; notice the Y shaped vertex just below the aiming dot.

![shot0009](https://user-images.githubusercontent.com/2976480/77261494-0f67da00-6c4c-11ea-80b2-e1c1215f6aca.jpg)

I encountered this with the Debian package, `1.51.c+dfsg1-2`, and I was able to reproduce it with git `2f4c32908497d0`.

Hardware:
* Ryzen 7 2700X
* 64 GB RAM
* GeForce GTX 460

Software:
* Debian Sid
* Linux kernel 5.4.0-3-amd64
* nvidia binary driver 390.132-1 (nvidia-legacy-390xx Debian packaging)

Initial environment:
1. pk3 files are from the GOG installer, `setup_return_to_castle_wolfenstein_2.0.0.2.exe`, installed via wine, then packaged using `game-data-packager`. Each pk3 file is symlinked into `/usr/local/games/wolf/main`.
2. `~/.wolf` is empty except for a saved game to illustrate the problem. See attachment:
[dot-wolf.tar.gz](https://github.com/iortcw/iortcw/files/4366429/dot-wolf.tar.gz)
(Note that contents of the tar are a `.wolf` directory, which will be "hidden" by default.)
3. wolfsp built and installed from git `2f4c32908497d0`, using `make -j16 copyfiles`.

Steps to reproduce:
1. `/usr/local/games/wolf/iowolfsp.x86_64 +set cl_renderer rend2 +bind F12 screenshotJPEG 2>&1 | tee /tmp/wolfsp.log`
2. Load `tmp` saved game.
3. Watch sky; hit F12 for screenshots as desired.

Observed behavior: three lines coming to a point, as shown in the screenshot. Depending on cloud movement, this can be hard to see.

Expected behavior: lines not visible.

Further logs:
* wolfsp.log (stdout and stderr from binary):
[wolfsp.log](https://github.com/iortcw/iortcw/files/4366438/wolfsp.log)
* output from glxinfo:
[glxinfo.txt](https://github.com/iortcw/iortcw/files/4366439/glxinfo.txt)

Please let me know if there is any other relevant information I can provide.

Thanks,
Corey

**Comment by KuehnhammerTobias** [comment]
Unfortunately I can't solve the issue you mentioned, but I can at least point you in the right direction, I think.
The problem is the fog, not the sky.
There was a new fogtype added to Return to Castle Wolfenstein which was coded in a way that isn't supported by the second/alternate renderer (at least for now).
You can try this by starting the map you mentioned, and turn off the fog by setting the r_wolffog cvar to 0.
The lines shouldn't be there anymore, unfortunately the fog is also gone.
I'm not sure if this causes the visual issue you experience, but I'm relatively sure this is the case.
So what can you do? Well, unfortunately to fix this issue you need to support the new fog types by making changes to the rendering code.
If you have programming skills (as far as rendering is concerned) you can eventually have a look at the zturtleman's Spearmint engine that supports those new fog types: https://github.com/zturtleman/spearmint/wiki/Shaders#nofog-origin-rtcw
Unfortunately you need programming skills (as far as renderering is concerned) to fix the issue this way.
So, I'm sorry that I can't really help you,...

**Comment by bugfood** [comment]
Thanks @KuehnhammerTobias , you're absolutely right about `r_wolffog`.

I took a look at the spearmint code, and I compared spearmint `code/renderergl2/tr_shade.c` to iortcw `SP/code/rend2/tr_shade.c`. The files seem to have diverged quite a bit from whatever common ancestor they have, and I think I'm overwhelmed.

Trying to port `RB_FogPass()` led to trying to copy in `ComputeFogValues()` entirely, but that has a different final parameter, which means that `RB_IterateStagesGeneric()` needs to change at least some, and the two implementations of that are very different....

At least, that's where I found myself. I understand C well enough to know the operation of bits of code at the small scale, but I don't know the codebase or graphics concepts well enough to see the big picture of what exactly needs to be ported.

If this is easier than it looks and somebody can give me some very specific newbie-oriented pointers, then I can make another attempt. Otherwise, I'll have to leave this to someone who knows that they are doing.

-Corey

**Comment by MAN-AT-ARMS** [comment]
The wolf fog is and has been an issue even since vanilla. I've never been able to get anywhere with it either.

**Comment by KuehnhammerTobias** [comment]
@MAN-AT-ARMS I'm sure if you ask zturtleman to apply his changes to r2 he will be okay with it.
You remember, he already fixed a few things concerning iortcw. Moreover he is a really very nice person.
The good thing is that his changes to r2 to support the wolf fog is compatible with the current, existing rtcw assets.
Well, you're the boss, I just think you should consider to use his fixes/enhancements. Though, I know that the changes that will be made to r2 would be huge (as far as changed code lines are concerned). On the other hand, renderer2 seems pretty dead, so I think it should be okay to update the renderer.
This is just my personal opinion...

**Comment by MAN-AT-ARMS** [comment]
You're right he is always the go to person for issues and has been invaluable. He does have commit access to the project so I'm sure he will if inclined to.
